tests/vcs/VCSCommitBearTest.py

```diff
@@ -0,0 +1,147 @@
+import os
+import platform
+import shutil
+import stat
+import unittest
+import unittest.mock
+from queue import Queue
+from tempfile import mkdtemp
+
+from coalib.testing.BearTestHelper import generate_skip_decorator
+from bears.vcs.VCSCommitBear import VCSCommitBear, CommitResult
+from bears.vcs.git.GitCommitBear import GitCommitBear
+from coalib.settings.Section import Section
+from coalib.misc.Shell import run_shell_command
+from .git.GitCommitBearTest import GitCommitBearTest
+
+
+@generate_skip_decorator(GitCommitBear)
+class AnalyzeGitCommitTest(unittest.TestCase):
+
+    def run_uut(self, *args, **kwargs):
+        """
+        Runs the unit-under-test (via `self.uut.run()`) and collects the
+        messages of the yielded results as a list.
+
+        :param args:   Positional arguments to forward to the run function.
+        :param kwargs: Keyword arguments to forward to the run function.
+        :return:       A list of result values that give information related
+                       to head commit.
+        """
+        return list((result.raw_commit_message,
+                     result.commit_sha,
+                     result.commit_type,
+                     result.added_files,
+                     result.modified_files,
+                     result.deleted_files) for result in self.uut.run(
+                     *args, **kwargs))
+
+    def setUp(self):
+        self.msg_queue = Queue()
+        self.section = Section('')
+        self.uut = GitCommitBear(None, self.section, self.msg_queue)
+
+        self._old_cwd = os.getcwd()
+        self.gitdir = mkdtemp()
+        os.chdir(self.gitdir)
+        GitCommitBearTest.run_git_command('init')
+        GitCommitBearTest.run_git_command(
+            'config', 'user.email coala@coala.io')
+        GitCommitBearTest.run_git_command('config', 'user.name coala')
+
+    @staticmethod
+    def _windows_rmtree_remove_readonly(func, path, excinfo):
+        os.chmod(path, stat.S_IWRITE)
+        func(path)
+
+    def tearDown(self):
+        os.chdir(self._old_cwd)
+        if platform.system() == 'Windows':
+            onerror = self._windows_rmtree_remove_readonly
+        else:
+            onerror = None
+        shutil.rmtree(self.gitdir, onerror=onerror)
+
+    def test_analyze_git_commit(self):
+        run_shell_command('git checkout -b testbranch master')
+        run_shell_command('touch testfile1.txt ')
```

Comment:
As @jayvdb said in https://github.com/coala/coala-bears/pull/2501#discussion_r199858579

Use https://docs.python.org/3/library/pathlib.html#pathlib.Path.touch